src/core/packages/ui-settings/server-internal/src/settings/theme.ts

```diff
@@ -12,32 +12,54 @@ import { i18n } from '@kbn/i18n';
 import type { ThemeVersion } from '@kbn/ui-shared-deps-npm';
 import {
   type UiSettingsParams,
+  type ThemeName,
   parseThemeTags,
   SUPPORTED_THEME_NAMES,
+  DEFAULT_THEME_NAME,
 } from '@kbn/core-ui-settings-common';
 
-function getThemeInfo(options: GetThemeSettingsOptions) {
-  if (options?.isDist ?? true) {
-    return {
-      defaultDarkMode: false,
-    };
-  }
+interface ThemeInfo {
+  defaultDarkMode: boolean;
+  defaultThemeName: ThemeName;
+}
 
+const getThemeInfo = ({ isDist, isServerless }: GetThemeSettingsOptions): ThemeInfo => {
   const themeTags = parseThemeTags(process.env.KBN_OPTIMIZER_THEMES);
-  return {
-    defaultDarkMode: themeTags[0].endsWith('dark'),
+
+  const themeInfo: ThemeInfo = {
+    defaultDarkMode: false,
+    defaultThemeName: DEFAULT_THEME_NAME,
   };
-}
 
-interface GetThemeSettingsOptions {
-  isDist?: boolean;
-  isThemeSwitcherEnabled?: boolean;
+  if (!isDist) {
+    // Allow environment-specific config when not building for distribution
+    themeInfo.defaultDarkMode = themeTags[0]?.endsWith('dark') || false;
+  }
+
+  if (!isServerless) {
+    // Default to Borealis theme in non-serverless
+    themeInfo.defaultThemeName = 'borealis';
```

Comment:
This feels brittle, depending on whether or not we're in serverless to default the theme.  I feel like this shouldn't be determined here.